llama-index-core/llama_index/core/agent/custom/simple.py

```diff
@@ -253,7 +253,7 @@ def _finalize_task(self, state: Dict[str, Any], **kwargs: Any) -> None:
     def finalize_task(self, task: Task, **kwargs: Any) -> None:
         """Finalize task, after all the steps are completed."""
         # add new messages to memory
-        task.memory.set(task.memory.get() + task.extra_state["memory"].get_all())
+        task.memory.put_messages(task.extra_state["new_memory"].get_all())
```

Comment:
Curious why this changed? `initialize_step` does not initialize a `new_memory` item

Reply:
Yeah, a bit confused here as well. I think maybe this is by mistake. Going to revert this back!